oasisriddle/docs/riddle_eng

```diff
@@ -0,0 +1,62 @@
+# Oasis Riddle
+
+Developers at [Oasis Network][oasis-network] have created a short riddle
+that unlocks the path to a ROSE treasure. They hid it on the [Oasis Sapphire][oasis-explorer]
+blockchain and your job now is to write a script connecting to the
+smart contracts already deployed on Sapphire that will enable you to obtain
+the secret questions. If you manage to answer all the questions correctly, 
+you'll be able to collect your ROSE treasure! 🌹
+
+## What if I don't exactly know how Blockchains, Ethereum, Solidity, Dapps work?
+
+If you know C, Python or Javascript, you have all the basics covered, 
+everything else you'll be able to learn during the challange. 🎓
+
+Blockchains are generally not a simple thing and developing dApps has lot 
+more limitations compared to desktop or web apps. That is why one needs 
+an open mind while also having great understanding of algorhythms, 
+data structures, writting effecient code, bug hunting skills, thorough
+understanding of the programs from a cryptographis and security point of view, etc. 🤓
+
+## OK. What do I need to do?
+
+1. First join our `#sapphire-paratime` channel on our official
+   [Oasis Discord server][oasis-discord] so if you get stuck, you can ask for help.
+   Even if you don't need help, stop by and say hi, everybody will be happy to meet you. 😉
+
+2. If this is your first time developing a dApp, you can watch a 
+   [short video on Ethereum smart contracts][ethereum-tutorial]. 💡
+
+3. When you're ready to move further, check out our 
+   [quickstart guide to Sapphire confidential smart contracs][oasis-sapphire-quickstart]. 💎
+
+4. After you're done with both tutorials, it's time to solve the riddle. You can find it at
+   `0x4874c1EC02c122Dfb29C994264dFCC4aD9FA8b96` on [Oasis Sapphire Testnet][oasis-explorer-testnet]. 
+   
+   The ABI of the contract is:
+
+```solidity
+function getQuestion(string memory coupon) external view returns (string memory)
+function submitAnswer(string memory coupon, string memory answer) external
+function claimReward(string memory coupon) external view returns (string memory)
+```
+
+5. Your task is to write a script with the help of the newly obtained skills and tolls, 
+   that communicates with the riddle's smart contract. First use a confidential call
+   `getQuestion()` to get the secret questions tied to your unique ID. Then create
+   a secret answer and send it back with a confidential transaciton via 
+   `submitAnswer()`. If the answer is correct, you'll be able to send a confidential 
+   `claimReward()` call in the next block to obtain the private key 
+   to your ROSE treasure on [Oasis Sapphire Mainnet][oasis-explorer]. 🎉
+
+6. Congratulations, you've unlocked your ROSE treasure! You can now use your ROSE coins 
+   to create new dApps on Oasis and be among the first to utilize confidential smart contracts. 
+   Or you can try to cash them in for a beer with your friends! 🍻
```

Comment:
?

```suggestion
   Or you can try to cash them in for a [beer with your friends][bitcoin-map]! 🍻
```

Reply:
The easiest way to cash them out is through an exchange but don't want to promote that, going from ROSE to bitcoin, you probably don't get anything out :D That's why I wrote just try to cash them out for a beer without any suggestions